awx/main/scheduler/task_manager.py

```diff
@@ -267,185 +256,47 @@ class DependencyManager(TaskBase):
-            if self.should_update_related_project(inventory_task, latest_src_project_update):
-                latest_src_project_update = self.create_project_update(inventory_task, project_id=invsrc.source_project_id)
-                created_dependencies.append(latest_src_project_update)
-            self.add_dependencies(inventory_task, [latest_src_project_update])
-            latest_src_project_update.scm_inventory_updates.add(inventory_task)
-        return created_dependencies
-
-    @timeit
-    def generate_dependencies(self, undeped_tasks):
-        created_dependencies = []
-        for task in undeped_tasks:
-            task.log_lifecycle("acknowledged")
-            if type(task) is Job:
-                created_dependencies += self.gen_dep_for_job(task)
-            elif type(task) is InventoryUpdate:
-                created_dependencies += self.gen_dep_for_inventory_update(task)
-            else:
-                continue
-        UnifiedJob.objects.filter(pk__in=[task.pk for task in undeped_tasks]).update(dependencies_processed=True)
-
-        return created_dependencies
-
-    def process_tasks(self):
-        deps = self.generate_dependencies(self.all_tasks)
-        self.generate_dependencies(deps)
-        self.subsystem_metrics.inc(f"{self.prefix}_pending_processed", len(self.all_tasks) + len(deps))
-
     @timeit
     def _schedule(self):
-        self.get_tasks(dict(status__in=["pending"], dependencies_processed=False))
-
-        if len(self.all_tasks) > 0:
-            self.get_inventory_source_tasks()
-            self.process_tasks()
+        failing_job_qs = UnifiedJob.objects.filter(
+            status='pending', dependencies_processed=False, dependent_jobs__status__in=['failed', 'error']
+        ).prefetch_related('dependent_jobs')
+        failed_job_ct = 0
+        already_failed = set()
+        for task in failing_job_qs.iterator():
+            messages = []
+            for dep in task.dependent_jobs.all():
+                if dep.status in ('failed', 'error') or dep.id in already_failed:
+                    messages.append(
+                        'Previous Task Failed: {"job_type": "%s", "job_name": "%s", "job_id": "%s"}'
+                        % (
+                            get_type_for_model(type(dep)),
+                            dep.name,
+                            dep.id,
+                        )
+                    )
+            # if we detect a failed or error dependency, go ahead and fail this
+            # task. The errback on the dependency takes some time to trigger,
+            # and we don't want the task to enter running state if its
+            # dependency has failed or errored.
+            task.status = 'failed'
+            task.job_explanation = '\n'.join(messages)
+            task.save(update_fields=['status', 'job_explanation'])
+            task.websocket_emit_status('failed')
+            failed_job_ct += 1
+            already_failed.add(task.id)
```

Comment:
Since I moved this stuff for failing jobs, we could separately track the timing and counts in metrics. This should _fairly rarely_ actually process anything, but it is very logically distinct from the other part (advancing jobs).